ORM/org.framed.orm.transformation/epsilon/Relationship.etl

```diff
@@ -1,3 +1,31 @@
+operation Any createPlace(referenced : source!ModelElement, label : source!NamedElement) : target!Place {
+	var place = new target!Place;
+	
+	place.holder = referenced.equivalent();
```

Comment:
Hmm,

We could use the referencedRoles reference of relations to get holder
working.
But we need to ensure that only RoleTypes are referenced then.

Otherwise we could use the targetLabel and sourceLabel reference, too.
This is possible because a NamedModelElement is referenced there and
RoleTypes are NamedModelElements...

What do you guys think?

Cheers
Johannes

2015-01-15 20:49 GMT+01:00 Tobias Nett <notifications@github.com>:

> In ORM/org.framed.orm.transformation/epsilon/Relationship.etl
> <https://github.com/leondart/FRaMED/pull/74#discussion-diff-23037804>:
>
> > @@ -1,3 +1,31 @@
> > +operation Any createPlace(referenced : source!ModelElement, label : source!NamedElement) : target!Place {
> > +	var place = new target!Place;
> > +	
> > +	place.holder = referenced.equivalent();
>
> This can be limited to referenced RoleTypes easily:
>
> if (referenced.getType.getValue == 2) {
>   ..
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/leondart/FRaMED/pull/74/files#r23037804>.
>